packages/amplify-category-auth/resources/auth-custom-resource/hostedUILambda.js

```diff
@@ -1,81 +1,73 @@
 const response = require('cfn-response');
 const aws = require('aws-sdk');
 const identity = new aws.CognitoIdentityServiceProvider();
-exports.handler = (event, context, callback) => {
-  const userPoolId = event.ResourceProperties.userPoolId;
-  const inputDomainName = event.ResourceProperties.hostedUIDomainName;
-  let deleteUserPoolDomain = (domainName) => {
-    let params = { Domain: domainName, UserPoolId: userPoolId };
-    return identity.deleteUserPoolDomain(params).promise();
+
+exports.handler = (event, context) => {
+  // Don't return promise, response.send() marks context as done internally
+  const ignoredPromise = handleEvent(event, context);
+};
+
+async function checkDomainAvailability(domainName) {
+  const params = { Domain: domainName };
+  try {
+    const res = await identity.describeUserPoolDomain(params).promise();
+    if (res.DomainDescription && res.DomainDescription.UserPool) {
+      return false;
+    }
+    return true;
```

Comment:
nit
```suggestion
    return !(res.DomainDescription && res.DomainDescription.UserPool)
```

Reply:
@0618 I'll be working on migrating this lambda to Node18 and SDK V3 in follow PR in ~a week. 
Mind if I address all these nits there ? Re-recording snapshots is time consuming.

Reply:
Sure. Those are not blockers. Just nice to have